--- a/tests/test-coredump-unwind.c	2012-10-06 06:54:38.000000000 +0200
+++ b/tests/test-coredump-unwind.c	2014-12-17 21:13:25.073170288 +0100
@@ -57,7 +57,9 @@
 #include <grp.h>
 
 /* For SIGSEGV handler code */
-#include <execinfo.h>
+#if HAVE_EXECINFO_H
+  extern int backtrace (void **, int);
+#endif
 #include <sys/ucontext.h>
 
 #include <libunwind-coredump.h>
@@ -242,11 +242,11 @@
 			ip);
 
   {
+#if defined(__linux__) && HAVE_EXECINFO_H
     /* glibc extension */
     void *array[50];
     int size;
     size = backtrace(array, 50);
-#ifdef __linux__
     backtrace_symbols_fd(array, size, 2);
 #endif
   }
