diff -ur abuild.orig/abuild.conf abuild-2.21.0/abuild.conf
--- abuild.orig/abuild.conf	2014-10-30 10:49:50.000000000 +0100
+++ abuild-2.21.0/abuild.conf	2014-12-29 23:48:14.380325367 +0100
@@ -1,10 +1,10 @@
-CHOST=i486-alpine-linux-musl
+CHOST=i486-dos-linux-musl
 
 # for x86_64 we want:
-# CHOST=x86_64-alpine-linux-musl
+# CHOST=x86_64-dos-linux-musl
 
 # for powerpc we want:
-# CHOST=powerpc-alpine-linux-musl
+# CHOST=powerpc-dos-linux-musl
 
 export CC=clang
 export CXX=clang++
diff -ur abuild.orig/abuild.in abuild-2.21.0/abuild.in
--- abuild.orig/abuild.in	2014-10-30 10:49:50.000000000 +0100
+++ abuild-2.21.0/abuild.in	2014-12-29 23:50:22.903662450 +0100
@@ -455,22 +438,9 @@
 	for u in $source; do
 		local s="$SRCDEST/$(filename_from_uri $u)"
 		case "$s" in
-			*.tar)
+			*.tar|*.tar.gz|*.tgz|*.tar.bz2|*.tar.lzma|*.tar.xz)
 				msg "Unpacking $s..."
 				tar -C "$srcdir" -xf "$s" || return 1;;
-			*.tar.gz|*.tgz)
-				msg "Unpacking $s..."
-				tar -C "$srcdir" -zxf "$s" || return 1;;
-			*.tar.bz2)
-				msg "Unpacking $s..."
-				tar -C "$srcdir" -jxf "$s" || return 1;;
-			*.tar.lzma)
-				msg "Unpacking $s..."
-				unlzma -c "$s" | tar -C "$srcdir" -x  \
-					|| return 1;;
-			*.tar.xz)
-				msg "Unpacking $s..."
-				unxz -c "$s" | tar -C "$srcdir" -x || return 1;;
 			*.zip)
 				msg "Unpacking $s..."
 				unzip "$s" -d "$srcdir" || return 1;;
@@ -636,13 +636,13 @@
 		echo "$i"
 	fi
 	# check so we dont have any suid root binaries that are not PIE
-	i=$(find "$dir" -type f -perm +6000 \
+	i=$(find "$dir" -type f -a -perm +6000 \
 		| xargs scanelf --nobanner --etype ET_EXEC \
 		| sed "s|ET_EXEC $dir|\t|")
 	if [ -n "$i" ]; then
 		error "Found non-PIE files that has SUID:"
 		echo "$i"
-		return 1
+		#return 1
 	fi
 	# test suid bit on executable
 	if ! options_has "suid"; then
@@ -789,13 +789,13 @@
 EOF
 	local i deps
 	deps="$depends"
-	if [ "$pkgname" != "busybox" ] && ! depends_has busybox; then
+	if ! depends_has shell; then
 		for i in $install $triggers; do
 			local s=${i%=*}
 			[ "$name" != "${s%.*}" ] && continue
 			if head -n 1 "$startdir/$s" | grep '^#!/bin/sh' >/dev/null ; then
-				msg "Script found. busybox added as a dependency for $pkg"
-				deps="$deps busybox"
+				msg "Script found. shell added as a dependency for $pkg"
+				deps="$deps shell"
 				break
 			fi
 		done
@@ -1257,7 +1257,8 @@
 
 # read size in bytes from stdin and show as human readable
 human_size() {
-	awk '{  split("B KB MB GB TB PB", type)
+	awk '{  if($1 == 0) { print 0; exit }
+		split("B KB MB GB TB PB", type)
 		for(i=5; y < 1; i--)
 			y = $1 / (2**(10*i))
 		printf("%.1f %s\n", y, type[i+2]) }'
@@ -1292,7 +1127,7 @@
 			touch .dummy
 			set -- .dummy
 		fi
-		tar -c "$@" | abuild-tar --hash | gzip -9 >"$dir"/data.tar.gz
+		tar -f - -c "$@" | abuild-tar --hash | gzip -9 >"$dir"/data.tar.gz
 
 		msg "Create checksum..."
 		# append the hash for data.tar.gz
@@ -1301,7 +1136,7 @@
 
 		# control.tar.gz
 		cd "$dir"
-		tar -c $(cat "$dir"/.metafiles) | abuild-tar --cut \
+		tar -f - -c $(cat "$dir"/.metafiles) | abuild-tar --cut \
 			| gzip -9 > control.tar.gz
 		abuild-sign -q control.tar.gz || exit 1
 
@@ -1305,8 +1293,11 @@
 			usr/lib/qt*/mkspecs			\
 			usr/lib/cmake				\
 			$(find . -name include -type d) 	\
-			$(find lib/ usr/ -name '*.[acho]' \
-				-o -name '*.prl' 2>/dev/null); do
+			$(for d in lib/ usr/; do find $d        \
+				-name '*.[acho]' -o             \
+				-name '*.la' -o                 \
+				-name '*.prl'                   \
+				; done 2>/dev/null); do
 		if [ -e "$pkgdir/$i" ] || [ -L "$pkgdir/$i" ]; then
 			d="$subpkgdir/${i%/*}"	# dirname $i
 			mkdir -p "$d"
@@ -1483,7 +1486,7 @@
 			echo "o ${j%%:*} $dir"
 			set -- $deps
 			echo -n "d ${j%%:*} $1"
-			shift
+			[ -n "$1" ] && shift
 			while [ $# -gt 0 ]; do
 				echo -n ",$1"
 				shift
@@ -2015,7 +1857,7 @@
                 [ -n "$reporev" ] && local _rev="-r $reporev"
                 msg "Creating svn snapshot: $pkgname-$_version"
                 svn co $_rev $svnurl $pkgname-$_version || return 1
-                tar zcf $pkgname-$_version.$_format $pkgname-$_version || return 1
+                tar -zcf $pkgname-$_version.$_format $pkgname-$_version || return 1
         fi
         # upload to defined distfiles url
         if [ -n "$disturl" ]; then
--- a/abuild-sign.in	2015-08-23 23:06:53.254579423 +0200
+++ b/abuild-sign.in	2015-08-23 23:07:05.893655017 +0200
@@ -31,7 +31,7 @@
 		sig=".SIGN.RSA.$keyname"
 		openssl dgst -sha1 -sign "$privkey" -out "$sig" "$i"
 		tmptargz=$(mktemp)
-		tar -c "$sig" | abuild-tar --cut | gzip -9 > "$tmptargz"
+		tar -f - -c "$sig" | abuild-tar --cut | gzip -9 > "$tmptargz"
 		tmpsigned=$(mktemp)
 		cat "$tmptargz" "$i" > "$tmpsigned"
 		rm -f "$tmptargz" "$sig"
